File: core/open_childhood.go
// OpenChildhood -- Toda Crianca e da Republica -- gerado de Portugol++
package openchildhood_toda_crianca_e_da_republica

import "fmt"

// !/usr/bin/env python3
//
OpenChildhood -- Toda Crianca && da Republica
===============================================
"Na Republica, nenhum bebe nasce 'privado'.
Nenhum bebe tem 'pai devedor'.
Toda crianca && responsabilidade COLETIVA.
A pensao ! existe porque a dependencia ! existe.
O individuo amadurece sustentado por TODOS."
O PROBLEMA DA PENSÃO:
A pensao alimenticia existe porque o sistema capitalista individualiza
a responsabilidade de criar um ser humano em 2 pessoas (pai + mae).
Mas criar uma crianca exige:
- Comida, agua, moradia, saude, educacao (direitos garantidos pela Republica)
- Tempo, afeto, presenca, estimulo (isso SIM && dos pais/comunidade)
- Protecao, orientacao, mentoria (comunidade + OpenFamily)
Na Republica:
- Comida, moradia, saude sao DIREITOS. Ninguem precisa "pagar pensao"
    para que filho tenha o que ja e direito universal. {
- O trabalho de CUIDAR (tempo, afeto) && CONTRIBUTIÇÃO reconhecida.
- Quem cuida de crianca recebe credito de impacto (multiplicador geracional).
RESULTADO:
- Zero pensao (! ha dinheiro)
- Zero abandono (comunidade garante cuidado)
- Zero "mae solo" (! ha "solo" -- comunidade cria junto)
- Zero "pai ausente" (ausencia financeira ! existe; ausencia afetiva
    && tratada por OpenFamily + OpenPsychology)
Author: OpenRepublic Team
//
// importa annotations de __future__
// importa math
// importa time
// importa dataclass, field de dataclasses
// importa Any, Dict, List, Optional, Tuple de typing
// importa Enum de enum
// importa defaultdict de collections
// ============================================================================
// Childhood Stages
// ============================================================================
type DevelopmentStage int
const (
    INFANT = "bebe_0_2"  // 0-2 anos: cuidado 24h
    TODDLER = "primeira_infancia_2_5"  // 2-5: estimulo motor/linguagem
    CHILD = "infancia_5_12"  // 5-12: educacao fundamental
    ADOLESCENT = "adolescente_12_18"  // 12-18: identidade + habilidades
    YOUNG_ADULT = "jovem_adulto_18_25"  // 18-25: transicao para autonomia
type CareRole int
const (
    // Quem cuida da crianca -- todos com igual valor.
    BIOLOGICAL_PARENT = "pai_biologico"
    CHOSEN_PARENT = "pai_escolhido"
    COMMUNITY_PARENT = "parente_comunitario"
    ELDER_MENTOR = "mentor_idoso"
    EDUCATOR = "educador"
    HEALTH_PROVIDER = "profissional_saude"
    PEER_MENTOR = "mentor_mais_velho"
    REPUBLIC = "propria_republica"
type SupportType int
const (
    // O que a Republica garante para toda crianca.
    // Materiais (DIREITO -- zero credito, zero pensao)
    FOOD = "alimentacao"
    HOUSING = "moradia"
    HEALTHCARE = "saude"
    EDUCATION = "educacao"
    CLOTHING = "vestuario"
    SAFETY = "seguranca"
    // Emocionais (COMUNIDADE + familia)
    LOVE = "amor_presenca"
    STIMULATION = "estimulo_cognitivo"
    SOCIAL = "convivio_social"
    PROTECTION = "protecao"
// decorador: @dataclass
type ChildProfile struct {
    // Perfil de uma crianca na Republica.
    child_id: texto
    name: texto
    age_years: flutuante
    stage: DevelopmentStage
    // Quem cuida (multiplas pessoas, igual valor)
    caregivers := field(default_factory=list) // ids // [texto]
    // O que recebe
    material_needs_met := true // SEMPRE (direito) // bool
    emotional_needs_score := 80.0 // 0-100 (monitorado) // float64
    social_integration := true // bool
    education_engaged := true // bool
    health_status := "saudavel" // string
    // Desenvolvimento
    developmental_milestones := field(default_factory=dict) // {texto: logico}
    // Vulnerabilidades
    at_risk := false // bool
    risk_factors := field(default_factory=list) // [texto]
    interventions := field(default_factory=list) // [texto]
type ChildhoodSystem struct {
    // Sistema que garante maturidade de todo individuo.
    A Republica SE RESPONSABILIZA por:
    1. Toda necessidade MATERIAL (comida, moradia, saude, educacao)
    -> Ninguem precisa "pagar pensao" -- && direito universal
    2. Toda necessidade DESENVOLVIMENTAL (estimulo, educacao)
    -> OpenEducation (Feynman + KISS + fazer)
    3. Toda necessidade EMOCIONAL (amor, presenca, comunidade)
    -> OpenFamily + OpenSocial + comunidade
    4. Toda necessidade de PROTECAO
    -> OpenFamily protection system
    O QUE ! EXISTE MAIS:
    - Pensao alimenticia (! ha dinheiro)
    - "Mae solo" (! ha "solo" -- comunidade cria)
    - "Pai ausente" financeiro (ausencia financeira impossivel)
    - Crianca passando fome porque pai "! paga"
    - Crianca sem escola porque mae "! tem dinheiro"
    - Divorci brigando por "guarda" (! ha propriedade de filho)
    - Crianca como "mercadoria" de disputa juridica
    O QUE EXISTE:
    - Crianca com TODAS necessidades materiais garantidas (direito)
    - Crianca com multiplos cuidadores (familia + comunidade)
    - Cuidar de crianca = CONTRIBUICAO de alto impacto
    - Ausencia AFETIVA tratada (! financeira)
    - Protecao infantil garantida por 3 camadas (OpenFamily)
    //
    func __init__(self) {
        self.children: {texto: ChildProfile} = {}
        self.caregiver_impact: {texto: flutuante} = defaultdict(flutuante)
        self._counter = 0
    func register_child(self, name: texto, age_years: flutuante) ChildProfile {
        self._counter += 1
        cid = "CHILD-{self._counter:05d}"
        stage = self._stage_from_age(age_years)
        child = ChildProfile(
            child_id = cid, name=name, age_years=age_years, stage=stage,
            material_needs_met = true) // SEMPRE true na Republica
        self.children[cid] = child
        return child
    func assign_caregiver(self, child_id: texto, caregiver_id: texto) {
        child = self.children.get(child_id)
        if child {
            child.caregivers.append(caregiver_id)
    funcao calculate_care_impact(self, caregiver_id: texto,
                            children_count: inteiro,
                            hours_per_day: flutuante) -> {texto: qualquer}:
        // Calcular impacto de cuidar de criancas.
        Cuidar de criancas tem impacto INTERGERACIONAL:
        - Cada crianca cuidada = 1 vida influenciada
        - Ripple factor 10x (crianca ensina outros no futuro)
        - Usar a formula do OpenCredit (impacto = pessoas x ripple)
        Quem cuida de crianca recebe ALTO credito de impacto
        porque a contribuicao afeta gerações futuras.
        //
        impact = hours_per_day * children_count * 10 // ripple 10x
        self.caregiver_impact[caregiver_id] += impact
        return {
            "caregiver": caregiver_id,
            "children": children_count,
            "hours_day": hours_per_day,
            "ripple_factor": 10,
            "impact_score": arredonde(impact, 1),
            "credit_weight": "ALTO (intergeracional)",
            "message": ("Cuidar de {children_count} crianca(s) por {hours_per_day}h/dia "
                    "= impacto {impact:.0f} (ripple 10x intergeracional). "
                    "Maior peso de credito da Republica."),
        }
    func assess_child(self, child_id: texto) {texto: qualquer} {
        // Avaliar bem-estar de uma crianca.
        child = self.children.get(child_id)
        if ! child {
            return {"error": "! encontrada"}
        // Material: SEMPRE garantido na Republica
        material = "GARANTIDO (direito universal)"
        // Emocional: depende de cuidadores + comunidade
        if len(child.caregivers) >= 2 {
            emotional = child.emotional_needs_score >= 80 ? "FORTÍSSIMO" : "ADEQUADO"
        } else if len(child.caregivers) >= 1 {
            emotional = child.emotional_needs_score >= 60 ? "ADEQUADO" : "ATENCAO"
        } else {
            emotional = "CRÍTICO: sem cuidador identificado"
            child.at_risk = true
        // Social
        social = child.social_integration ? "INTEGRADO" : "ISOLADO (risco)"
        // Education
        education = child.education_engaged ? "ENGAJADO" : "DESFASE"
        return {
            "child": child.name,
            "age": child.age_years,
            "stage": child.stage.value,
            "caregivers": len(child.caregivers),
            "material_needs": material,
            "emotional_needs": emotional,
            "social": social,
            "education": education,
            "health": child.health_status,
            "at_risk": child.at_risk,
            "verdict": self._verdict(child, emotional, social, education),
        }
    // decorador: @staticmethod
    func _stage_from_age(age: flutuante) DevelopmentStage {
        if age <= 2 {
            return DevelopmentStage.INFANT
        if age <= 5 {
            return DevelopmentStage.TODDLER
        if age <= 12 {
            return DevelopmentStage.CHILD
        if age <= 18 {
            return DevelopmentStage.ADOLESCENT
        return DevelopmentStage.YOUNG_ADULT
    // decorador: @staticmethod
    funcao _verdict(child: ChildProfile, emotional: texto,
                social: texto, education: texto) -> texto:
        if child.at_risk  ||  "CRÍTICO" in emotional {
            return "INTERVENCAO IMEDIATA: OpenFamily + comunidade"
        if "ATENCAO" in emotional  ||  "ISOLADO" in social {
            return "MONITORAR: oferecer apoio comunitario"
        return "DESENVOLVIMENTO SAUDAVEL"
// ============================================================================
// Main
// ============================================================================
if __name__ == "__main__" {
    fmt.Println("=" * 80)
    fmt.Println("  OPENCHILDHOOD -- TODA CRIANCA E DA REPUBLICA")
    fmt.Println("  'Nenhum bebe nasce privado. Toda crianca && coletiva.'")
    fmt.Println("=" * 80)
    system = ChildhoodSystem()
    // === 1. Register Children ===
    fmt.Println("\n\n  === CRIANCAS REGISTRADAS ===\n")
    children_data = [
        ("Joaozinho", 8, ["C-001", "C-002", "C-005 (avô comunitário)"]),
        ("Mariazinha", 5, ["C-001", "C-002"]),
        ("Pedrinho", 2, ["C-003", "C-006"]),
        ("Aninha", 14, ["C-004", "C-007 (mentor idoso)"]),
        ("Garoto Sem Cuidador", 10, []),   // caso de risco simulado
    ]
    para name, age, caregivers in children_data: {
        child = system.register_child(name, age)
        for _, cg := range caregivers {
            // Extract just IDs
            cg_id = cg.split(" ")[0]
            system.assign_caregiver(child.child_id, cg_id)
        fmt.Println("  {child.name} ({child.age_years} anos) - {child.stage.value}")
        fmt.Println("    Cuidadores: {len(child.caregivers)}")
        fmt.Println("    Necessidades materiais: {'GARANTIDAS' if child.material_needs_met else 'FALTANDO'}")
    // === 2. Assess ===
    fmt.Println("\n\n  === AVALIACAO ===\n")
    for _, cid := range system.children {
        result = system.assess_child(cid)
        fmt.Println("\n  {result['child']} ({result['age']} anos)")
        fmt.Println("    Material: {result['material_needs']}")
        fmt.Println("    Emocional: {result['emotional_needs']}")
        fmt.Println("    Social: {result['social']}")
        fmt.Println("    Educacao: {result['education']}")
        fmt.Println("    Risco: {'SIM' if result['at_risk'] else '!'}")
        fmt.Println("    Veredito: {result['verdict']}")
    // === 3. Care Impact ===
    fmt.Println("\n\n  === IMPACTO DE CUIDAR (credito) ===\n")
    impacts = [
        ("C-001", 2, 4, "Cleiton cuida 2 criancas 4h/dia"),
        ("C-002", 3, 6, "Amina cuida 3 criancas (creche comunitaria) 6h/dia"),
        ("C-005", 5, 8, "Lars (avô comunitario) cuida 5 criancas 8h/dia"),
        ("C-003", 1, 2, "Sven cuida 1 crianca 2h/dia"),
    ]
    para cg_id, n_child, hours, desc in impacts: {
        result = system.calculate_care_impact(cg_id, n_child, hours)
        fmt.Println("\n  {desc}")
        fmt.Println("    Impacto: {result['impact_score']} (ripple {result['ripple_factor']}x)")
        fmt.Println("    Peso credito: {result['credit_weight']}")
    // === 4. Why No Pension ===
    fmt.Println("\n\n{'='*80}")
    fmt.Println("  POR QUE A PENSAO NAO EXISTE MAIS")
    pensão_inexistente = "pensão alimentícia"
    fmt.Println("{'='*80}")
    fmt.Println("""
CAPITALISMO OPENREPUBLIC
--------------------------------------- ---------------------------------------
Pensao alimenticia ($/mes) ZERO (! ha dinheiro)
Pai "devedor" de pensao                  Pai ! deve dinheiro (! existe)
Mae "solo" (sozinha financeiramente)     Nao ha "solo" -- comunidade cria junto
Crianca passando fome se pai ! paga Comida && DIREITO (sempre garantida)
Crianca sem escola se mae ! paga Educacao && DIREITO (sempre garantida)
Divorci brigando por "guarda"            Nao ha "guarda" -- ! ha propriedade de filho
Filho como mercadoria juridica Filho && PESSOA com direitos
"Pai ausente" = ! paga                 "Pai ausente" = falta AFETO (tratado)
Justicia decide quanto filho custa Republica garante quanto filho precisa
Crianca refem de disputa financeira Crianca LIVRE de disputa
O QUE A REPUBLICA GARANTE PARA TODA CRIANCA:
    MATERIAL (DIREITO -- sempre, incondicional):
    - Comida: OpenFood
    - Moradia: OpenHome + OpenCivilConstruction
    - Saude: OpenHealth + OpenMedicine
    - Educacao: OpenEducation (Feynman + KISS + fazer)
    - Vestuario: OpenProduction (FabLab)
    - Seguranca: OpenFamily + OpenAutonomousSecurity
    EMOCIONAL (comunidade + familia):
    - Amor && presenca: multiplos cuidadores (OpenFamily)
    - Estimulo cognitivo: OpenEducation desde bebe
    - Convivio social: OpenSocial + comunidade local
    - Protecao: 3 camadas (familia -> comunidade -> sistema)
    DESENVOLVIMENTO (Republica garante maturidade):
    - Bebe (0-2): cuidado 24h (creche comunitaria)
    - Primeira infancia (2-5): estimulo motor + linguagem
    - Infancia (5-12): educacao fundamental (ensinar a FAZER)
    - Adolescente (12-18): identidade + habilidades
    - Jovem adulto (18-25): transicao para autonomia
CUIDAR DE CRIANCA = ALTO IMPACTO:
    Quem cuida de crianca recebe o MAIOR peso de credito.
    Ripple factor 10x (intergeracional).
    Cada crianca cuidada = vida influenciada para sempre.
    Cuidar de 3 criancas 6h/dia = impacto 180 (maior que cirurgia).
O QUE ACABA:
    - Pensao alimenticia
    - "Mae solo"
    - "Pai ausente" financeiro
    - Crianca refem de briga de divorcio
    - Crianca passando fome
    - Crianca sem escola
    - Crianca como propriedade
"Toda crianca && da Republica.
Nao de um pai. Nao de uma mae.
De TODOS.
&& TODOS garantem que ela cresca saudavel.
Comida, moradia, saude, educacao: direitos.
Amor, presenca, cuidado: comunidade.
Sem pensao. Sem disputa. Sem dinheiro.
Apenas cuidado."
// )

File: core/open_childhood.c
/* OpenChildhood -- Toda Crianca e da Republica -- gerado de Portugol++ */
#ifndef OPENCHILDHOOD_TODA_CRIANCA_E_DA_REPUBLICA_H
#define OPENCHILDHOOD_TODA_CRIANCA_E_DA_REPUBLICA_H

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

// !/usr/bin/env python3
//
OpenChildhood -- Toda Crianca && da Republica;
===============================================;
"Na Republica, nenhum bebe nasce 'privado'.;
Nenhum bebe tem 'pai devedor'.;
Toda crianca && responsabilidade COLETIVA.;
A pensao ! existe porque a dependencia ! existe.;
O individuo amadurece sustentado por TODOS.";
O PROBLEMA DA PENSÃO:;
A pensao alimenticia existe porque o sistema capitalista individualiza;
a responsabilidade de criar um ser humano em 2 pessoas (pai + mae).;
Mas criar uma crianca exige:;
- Comida, agua, moradia, saude, educacao (direitos garantidos pela Republica);
- Tempo, afeto, presenca, estimulo (isso SIM && dos pais/comunidade);
- Protecao, orientacao, mentoria (comunidade + OpenFamily);
Na Republica:;
- Comida, moradia, saude sao DIREITOS. Ninguem precisa "pagar pensao";
    /* para que filho tenha o que ja e direito universal. */
- O trabalho de CUIDAR (tempo, afeto) && CONTRIBUTIÇÃO reconhecida.;
- Quem cuida de crianca recebe credito de impacto (multiplicador geracional).;
RESULTADO:;
- Zero pensao (! ha dinheiro);
- Zero abandono (comunidade garante cuidado);
- Zero "mae solo" (! ha "solo" -- comunidade cria junto);
- Zero "pai ausente" (ausencia financeira ! existe; ausencia afetiva;
    && tratada por OpenFamily + OpenPsychology);
Author: OpenRepublic Team;
//
// importa annotations de __future__
// importa math
// importa time
// importa dataclass, field de dataclasses
// importa Any, Dict, List, Optional, Tuple de typing
// importa Enum de enum
// importa defaultdict de collections
// ============================================================================
// Childhood Stages
// ============================================================================
typedef struct DevelopmentStage {
    INFANT = "bebe_0_2"  // 0-2 anos: cuidado 24h;
    TODDLER = "primeira_infancia_2_5"  // 2-5: estimulo motor/linguagem;
    CHILD = "infancia_5_12"  // 5-12: educacao fundamental;
    ADOLESCENT = "adolescente_12_18"  // 12-18: identidade + habilidades;
    YOUNG_ADULT = "jovem_adulto_18_25"  // 18-25: transicao para autonomia;
typedef struct CareRole {
    // Quem cuida da crianca -- todos com igual valor.
    BIOLOGICAL_PARENT = "pai_biologico";
    CHOSEN_PARENT = "pai_escolhido";
    COMMUNITY_PARENT = "parente_comunitario";
    ELDER_MENTOR = "mentor_idoso";
    EDUCATOR = "educador";
    HEALTH_PROVIDER = "profissional_saude";
    PEER_MENTOR = "mentor_mais_velho";
    REPUBLIC = "propria_republica";
typedef struct SupportType {
    // O que a Republica garante para toda crianca.
    // Materiais (DIREITO -- zero credito, zero pensao)
    FOOD = "alimentacao";
    HOUSING = "moradia";
    HEALTHCARE = "saude";
    EDUCATION = "educacao";
    CLOTHING = "vestuario";
    SAFETY = "seguranca";
    // Emocionais (COMUNIDADE + familia)
    LOVE = "amor_presenca";
    STIMULATION = "estimulo_cognitivo";
    SOCIAL = "convivio_social";
    PROTECTION = "protecao";
// decorador: @dataclass
typedef struct ChildProfile {
    // Perfil de uma crianca na Republica.
    child_id: texto;
    name: texto;
    age_years: flutuante;
    stage: DevelopmentStage;
    // Quem cuida (multiplas pessoas, igual valor)
    [texto] caregivers = field(default_factory=list) // ids;
    // O que recebe
    bool material_needs_met = true // SEMPRE (direito);
    double emotional_needs_score = 80.0 // 0-100 (monitorado);
    bool social_integration = true;
    bool education_engaged = true;
    char* health_status = "saudavel";
    // Desenvolvimento
    {texto: logico} developmental_milestones = field(default_factory=dict);
    // Vulnerabilidades
    bool at_risk = false;
    [texto] risk_factors = field(default_factory=list);
    [texto] interventions = field(default_factory=list);
typedef struct ChildhoodSystem {
    // Sistema que garante maturidade de todo individuo.
    A Republica SE RESPONSABILIZA por:;
    1. Toda necessidade MATERIAL (comida, moradia, saude, educacao);
    -> Ninguem precisa "pagar pensao" -- && direito universal;
    2. Toda necessidade DESENVOLVIMENTAL (estimulo, educacao);
    -> OpenEducation (Feynman + KISS + fazer);
    3. Toda necessidade EMOCIONAL (amor, presenca, comunidade);
    -> OpenFamily + OpenSocial + comunidade;
    4. Toda necessidade de PROTECAO;
    -> OpenFamily protection system;
    O QUE ! EXISTE MAIS:;
    - Pensao alimenticia (! ha dinheiro);
    - "Mae solo" (! ha "solo" -- comunidade cria);
    - "Pai ausente" financeiro (ausencia financeira impossivel);
    - Crianca passando fome porque pai "! paga";
    - Crianca sem escola porque mae "! tem dinheiro";
    - Divorci brigando por "guarda" (! ha propriedade de filho);
    - Crianca como "mercadoria" de disputa juridica;
    O QUE EXISTE:;
    - Crianca com TODAS necessidades materiais garantidas (direito);
    - Crianca com multiplos cuidadores (familia + comunidade);
    - Cuidar de crianca = CONTRIBUICAO de alto impacto;
    - Ausencia AFETIVA tratada (! financeira);
    - Protecao infantil garantida por 3 camadas (OpenFamily);
    //
    void __init__(self) {
        self.children: {texto: ChildProfile} = {};
        self.caregiver_impact: {texto: flutuante} = defaultdict(flutuante);
        self._counter = 0;
    ChildProfile register_child(self, name: texto, age_years: flutuante) {
        self._counter += 1;
        cid = "CHILD-{self._counter:05d}";
        stage = self._stage_from_age(age_years);
        child = ChildProfile(;
            child_id = cid, name=name, age_years=age_years, stage=stage,;
            material_needs_met = true) // SEMPRE true na Republica;
        self.children[cid] = child;
        return child;
    void assign_caregiver(self, child_id: texto, caregiver_id: texto) {
        child = self.children.get(child_id);
        if (child) {
            child.caregivers.append(caregiver_id);
    funcao calculate_care_impact(self, caregiver_id: texto,
                            children_count: inteiro,;
                            hours_per_day: flutuante) -> {texto: qualquer}:;
        // Calcular impacto de cuidar de criancas.
        Cuidar de criancas tem impacto INTERGERACIONAL:;
        - Cada crianca cuidada = 1 vida influenciada;
        - Ripple factor 10x (crianca ensina outros no futuro);
        - Usar a formula do OpenCredit (impacto = pessoas x ripple);
        Quem cuida de crianca recebe ALTO credito de impacto;
        porque a contribuicao afeta gerações futuras.;
        //
        impact = hours_per_day * children_count * 10 // ripple 10x;
        self.caregiver_impact[caregiver_id] += impact;
        return {;
            "caregiver": caregiver_id,;
            "children": children_count,;
            "hours_day": hours_per_day,;
            "ripple_factor": 10,;
            "impact_score": arredonde(impact, 1),;
            "credit_weight": "ALTO (intergeracional)",;
            "message": ("Cuidar de {children_count} crianca(s) por {hours_per_day}h/dia ";
                    "= impacto {impact:.0f} (ripple 10x intergeracional). ";
                    "Maior peso de credito da Republica."),;
        };
    {texto: qualquer} assess_child(self, child_id: texto) {
        // Avaliar bem-estar de uma crianca.
        child = self.children.get(child_id);
        if (! child) {
            return {"error": "! encontrada"};
        // Material: SEMPRE garantido na Republica
        material = "GARANTIDO (direito universal)";
        // Emocional: depende de cuidadores + comunidade
        if (sizeof(child.caregivers) >= 2) {
            emotional = child.emotional_needs_score >= 80 ? "FORTÍSSIMO" : "ADEQUADO";
        } else if (sizeof(child.caregivers) >= 1) {
            emotional = child.emotional_needs_score >= 60 ? "ADEQUADO" : "ATENCAO";
        } else {
            emotional = "CRÍTICO: sem cuidador identificado";
            child.at_risk = true;
        // Social
        social = child.social_integration ? "INTEGRADO" : "ISOLADO (risco)";
        // Education
        education = child.education_engaged ? "ENGAJADO" : "DESFASE";
        return {;
            "child": child.name,;
            "age": child.age_years,;
            "stage": child.stage.value,;
            "caregivers": sizeof(child.caregivers),;
            "material_needs": material,;
            "emotional_needs": emotional,;
            "social": social,;
            "education": education,;
            "health": child.health_status,;
            "at_risk": child.at_risk,;
            "verdict": self._verdict(child, emotional, social, education),;
        };
    // decorador: @staticmethod
    DevelopmentStage _stage_from_age(age: flutuante) {
        if (age <= 2) {
            return DevelopmentStage.INFANT;
        if (age <= 5) {
            return DevelopmentStage.TODDLER;
        if (age <= 12) {
            return DevelopmentStage.CHILD;
        if (age <= 18) {
            return DevelopmentStage.ADOLESCENT;
        return DevelopmentStage.YOUNG_ADULT;
    // decorador: @staticmethod
    funcao _verdict(child: ChildProfile, emotional: texto,
                social: texto, education: texto) -> texto:;
        if (child.at_risk  ||  "CRÍTICO" in emotional) {
            return "INTERVENCAO IMEDIATA: OpenFamily + comunidade";
        if ("ATENCAO" in emotional  ||  "ISOLADO" in social) {
            return "MONITORAR: oferecer apoio comunitario";
        return "DESENVOLVIMENTO SAUDAVEL";
// ============================================================================
// Main
// ============================================================================
if (__name__ == "__main__") {
    printf("=" * 80);
    printf("  OPENCHILDHOOD -- TODA CRIANCA E DA REPUBLICA");
    printf("  'Nenhum bebe nasce privado. Toda crianca && coletiva.'");
    printf("=" * 80);
    system = ChildhoodSystem();
    // === 1. Register Children ===
    printf("\n\n  === CRIANCAS REGISTRADAS ===\n");
    children_data = [;
        ("Joaozinho", 8, ["C-001", "C-002", "C-005 (avô comunitário)"]),;
        ("Mariazinha", 5, ["C-001", "C-002"]),;
        ("Pedrinho", 2, ["C-003", "C-006"]),;
        ("Aninha", 14, ["C-004", "C-007 (mentor idoso)"]),;
        ("Garoto Sem Cuidador", 10, []),   // caso de risco simulado;
    ];
    /* para name, age, caregivers in children_data: */
        child = system.register_child(name, age);
        /* TODO: iterador C manual para cg em caregivers */
            // Extract just IDs
            cg_id = cg.split(" ")[0];
            system.assign_caregiver(child.child_id, cg_id);
        printf("  {child.name} ({child.age_years} anos) - {child.stage.value}");
        printf("    Cuidadores: {len(child.caregivers)}");
        printf("    Necessidades materiais: {'GARANTIDAS' if child.material_needs_met else 'FALTANDO'}");
    // === 2. Assess ===
    printf("\n\n  === AVALIACAO ===\n");
    /* TODO: iterador C manual para cid em system.children */
        result = system.assess_child(cid);
        printf("\n  {result['child']} ({result['age']} anos)");
        printf("    Material: {result['material_needs']}");
        printf("    Emocional: {result['emotional_needs']}");
        printf("    Social: {result['social']}");
        printf("    Educacao: {result['education']}");
        printf("    Risco: {'SIM' if result['at_risk'] else '!'}");
        printf("    Veredito: {result['verdict']}");
    // === 3. Care Impact ===
    printf("\n\n  === IMPACTO DE CUIDAR (credito) ===\n");
    impacts = [;
        ("C-001", 2, 4, "Cleiton cuida 2 criancas 4h/dia"),;
        ("C-002", 3, 6, "Amina cuida 3 criancas (creche comunitaria) 6h/dia"),;
        ("C-005", 5, 8, "Lars (avô comunitario) cuida 5 criancas 8h/dia"),;
        ("C-003", 1, 2, "Sven cuida 1 crianca 2h/dia"),;
    ];
    /* para cg_id, n_child, hours, desc in impacts: */
        result = system.calculate_care_impact(cg_id, n_child, hours);
        printf("\n  {desc}");
        printf("    Impacto: {result['impact_score']} (ripple {result['ripple_factor']}x)");
        printf("    Peso credito: {result['credit_weight']}");
    // === 4. Why No Pension ===
    printf("\n\n{'='*80}");
    printf("  POR QUE A PENSAO NAO EXISTE MAIS");
    pensão_inexistente = "pensão alimentícia";
    printf("{'='*80}");
    printf(""";
CAPITALISMO OPENREPUBLIC;
--------------------------------------- ---------------------------------------;
Pensao alimenticia ($/mes) ZERO (! ha dinheiro);
Pai "devedor" de pensao                  Pai ! deve dinheiro (! existe);
Mae "solo" (sozinha financeiramente)     Nao ha "solo" -- comunidade cria junto;
Crianca passando fome se pai ! paga Comida && DIREITO (sempre garantida);
Crianca sem escola se mae ! paga Educacao && DIREITO (sempre garantida);
Divorci brigando por "guarda"            Nao ha "guarda" -- ! ha propriedade de filho;
Filho como mercadoria juridica Filho && PESSOA com direitos;
"Pai ausente" = ! paga                 "Pai ausente" = falta AFETO (tratado);
Justicia decide quanto filho custa Republica garante quanto filho precisa;
Crianca refem de disputa financeira Crianca LIVRE de disputa;
O QUE A REPUBLICA GARANTE PARA TODA CRIANCA:;
    MATERIAL (DIREITO -- sempre, incondicional):;
    - Comida: OpenFood;
    - Moradia: OpenHome + OpenCivilConstruction;
    - Saude: OpenHealth + OpenMedicine;
    - Educacao: OpenEducation (Feynman + KISS + fazer);
    - Vestuario: OpenProduction (FabLab);
    - Seguranca: OpenFamily + OpenAutonomousSecurity;
    EMOCIONAL (comunidade + familia):;
    - Amor && presenca: multiplos cuidadores (OpenFamily);
    - Estimulo cognitivo: OpenEducation desde bebe;
    - Convivio social: OpenSocial + comunidade local;
    - Protecao: 3 camadas (familia -> comunidade -> sistema);
    DESENVOLVIMENTO (Republica garante maturidade):;
    - Bebe (0-2): cuidado 24h (creche comunitaria);
    - Primeira infancia (2-5): estimulo motor + linguagem;
    - Infancia (5-12): educacao fundamental (ensinar a FAZER);
    - Adolescente (12-18): identidade + habilidades;
    - Jovem adulto (18-25): transicao para autonomia;
CUIDAR DE CRIANCA = ALTO IMPACTO:;
    Quem cuida de crianca recebe o MAIOR peso de credito.;
    Ripple factor 10x (intergeracional).;
    Cada crianca cuidada = vida influenciada para sempre.;
    Cuidar de 3 criancas 6h/dia = impacto 180 (maior que cirurgia).;
O QUE ACABA:;
    - Pensao alimenticia;
    - "Mae solo";
    - "Pai ausente" financeiro;
    - Crianca refem de briga de divorcio;
    - Crianca passando fome;
    - Crianca sem escola;
    - Crianca como propriedade;
"Toda crianca && da Republica.;
Nao de um pai. Nao de uma mae.;
De TODOS.;
&& TODOS garantem que ela cresca saudavel.;
Comida, moradia, saude, educacao: direitos.;
Amor, presenca, cuidado: comunidade.;
Sem pensao. Sem disputa. Sem dinheiro.;
Apenas cuidado.";
// )

#endif // OPENCHILDHOOD_TODA_CRIANCA_E_DA_REPUBLICA_H
